cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -910,11 +910,19 @@ public Ingress generateExternalBootstrapIngress() {
     /**
      * Generates the annotations needed to configure the Ingress as TLS passthrough
      *
+     * @param ingressListener   The Ingress listener object with additional parameters and options
+     *
      * @return  Map with the annotations
      */
-    private Map<String, String> generateInternalIngressAnnotations() {
+    private Map<String, String> generateInternalIngressAnnotations(KafkaListenerExternalIngress ingressListener) {
         Map<String, String> internalAnnotations = new HashMap<>(4);
-        internalAnnotations.put("kubernetes.io/ingress.class", "nginx");
+
+        if (ingressListener.getIngressClass() != null) {
+            internalAnnotations.put("kubernetes.io/ingress.class", ingressListener.getIngressClass());
+        } else {
+            internalAnnotations.put("kubernetes.io/ingress.class", "nginx");
+        }
+
         internalAnnotations.put("ingress.kubernetes.io/ssl-passthrough", "true");
         internalAnnotations.put("nginx.ingress.kubernetes.io/ssl-passthrough", "true");
         internalAnnotations.put("nginx.ingress.kubernetes.io/backend-protocol", "HTTPS");
```

Comment:
just a question ... if the class isn't "nginx", do these annotations about nginx still make sense? Maybe not so we should not set them?

Reply:
We currently support only Nginx Ingress (mainly because the Ingress stuff is badly standardised in Kube). When you have Nginx Ingress, it will have the class `nginx` by default. But you can install multiple instances of Nginx Ingress for different purposes (for example each pointing to a different network, or a specific one for Kafka to separate it from HTTP traffic etc.) and configure it with different `class`. This is where this should be helpful.

Reply:
So if I got it right, different classes means different nginx classe so that the above annotations make always sense. Is that right?

Reply:
Yes, correct.